projects/gloo/cli/pkg/cmd/check/root.go

```
@@ -73,30 +81,37 @@ func RootCmd(opts *options.Options, optionsFunc ...cliutils.OptionsFunc) *cobra.
 			printer := printers.P{OutputType: opts.Top.Output}
 			printer.CheckResult = printer.NewCheckResult()
 
-			// TODO add back the kubegateway check (probably better to pass in as a flag rather than try to infer it)
-			// var err error
-			// call kubegateway check if k8s gateway crds are detected, otherwise call classic check
-			// isKubeGateway := detectKubeGatewayCrds(opts.Top.KubeContext)
-			// if isKubeGateway {
-			// 	err = kubegateway.Check(ctx, printer, opts)
-			// } else {
+			var multiErr *multierror.Error
+
+			// check edge gateway resources
 			err := CheckResources(ctx, printer, opts)
-			// }
+			if err != nil {
+				multiErr = multierror.Append(multiErr, err)
+			}
 
+			// check kube gateway resources
+			err = CheckKubeGatewayResources(ctx, printer, opts)
 			if err != nil {
-				// Not returning error here because this shouldn't propagate as a standard CLI error, which prints usage.
+				multiErr = multierror.Append(multiErr, err)
+			}
+
+			// check gloo fed resources (TODO this should return errors too)
```

wanted to make this similar to CheckResources and CheckKubeGatewayResources above and return errors (instead of just printing errors) but don't plan to handle this refactor in this PR